src/plugins/workflow_designer/src/library/DocWorkers.cpp

```diff
@@ -873,6 +875,134 @@ bool MSAWriter::isStreamingSupport() const {
+    data2document(doc, data, context);
+}
+
+void UgeneDBWriter::storeEntry(IOAdapter *io, const QVariantMap &data, int entryNum) {
+    streamingStoreEntry(format, io, data, context, entryNum);
+}
+
+void UgeneDBWriter::data2document(Document *doc, const QVariantMap &data, WorkflowContext *context) {
+    QScopedPointer<U2SequenceObject> seqObj(nullptr);
+    U2SequenceObject *dna = nullptr;
+    QString annotationName;
+
+    if (data.contains(BaseSlots::DNA_SEQUENCE_SLOT().getId())) {
+        SharedDbiDataHandler seqId = data[BaseSlots::DNA_SEQUENCE_SLOT().getId()].value<SharedDbiDataHandler>();
+        seqObj.reset(StorageUtils::getSequenceObject(context->getDataStorage(), seqId));
+        SAFE_POINT(nullptr != seqObj.data(), tr("UgeneDB writer: NULL sequence object"), );
+
+        U2OpStatusImpl os;
+        DNASequence seq = seqObj->getWholeSequence(os);
+        SAFE_POINT_OP(os, );
+
+        if (seq.getName().isEmpty()) {
+            int num = doc->findGObjectByType(GObjectTypes::SEQUENCE).size();
+            seq.setName(QString("unknown sequence %1").arg(num));
+        } else {
+            annotationName = getAnnotationName(seq.getName());
+        }
+
+        dna = qobject_cast<U2SequenceObject *>(doc->findGObjectByName(seq.getName()));
+        if (!dna && !seq.isNull()) {
+            dna = addSeqObject(doc, seq);
+        }
+    }
+
+    if (data.contains(BaseSlots::ANNOTATION_TABLE_SLOT().getId())) {
+        const QVariant &annsVar = data[BaseSlots::ANNOTATION_TABLE_SLOT().getId()];
+        const QList<SharedAnnotationData> atl = StorageUtils::getAnnotationTable(context->getDataStorage(), annsVar);
+
+        if (!atl.isEmpty()) {
+            AnnotationTableObject *att = nullptr;
+            if (dna) {
+                QList<GObject *> relAnns = GObjectUtils::findObjectsRelatedToObjectByRole(dna,
+                                                                                          GObjectTypes::ANNOTATION_TABLE,
+                                                                                          ObjectRole_Sequence,
+                                                                                          doc->getObjects(),
+                                                                                          UOF_LoadedOnly);
+                att = relAnns.isEmpty() ? nullptr : qobject_cast<AnnotationTableObject *>(relAnns.first());
+            }
+            if (!att) {
+                if (annotationName.isEmpty()) {
+                    int featuresNum = doc->findGObjectByType(GObjectTypes::ANNOTATION_TABLE).size();
+                    annotationName = QString("unknown features %1").arg(featuresNum);
+                }
+                att = qobject_cast<AnnotationTableObject *>(doc->findGObjectByName(annotationName));
+                if (att == nullptr) {
+                    doc->addObject(att = new AnnotationTableObject(annotationName, context->getDataStorage()->getDbiRef()));
```

Comment:
nit: do not use inlined initialization except some very common patterns like "if (auto x= cast<>)"